vscode-dotnet-runtime-library/src/Acquisition/ExistingPathResolver.ts

```diff
@@ -36,21 +66,121 @@ export class ExistingPathResolver
+                    return null;
                 }
             }
         }
+
+        return null;
+    }
+
+    private allowInvalidPath(workerContext : IAcquisitionWorkerContext) : boolean
+    {
+        return workerContext.allowInvalidPathSetting ?? false;
+    }
+
+    private async providedPathMeetsAPIRequirement(workerContext : IAcquisitionWorkerContext, existingPath : string, apiRequest : IDotnetAcquireContext) : Promise<boolean>
+    {
+
+        const availableRuntimes = await this.getRuntimes(existingPath);
+        const requestedMajorMinor = versionUtils.getMajorMinor(apiRequest.version, this.workerContext.eventStream, this.workerContext);
+
+        if(availableRuntimes.some((runtime) =>
+        {
+            return runtime.mode === apiRequest.mode && versionUtils.getMajorMinor(runtime.version, this.workerContext.eventStream, this.workerContext) === requestedMajorMinor;
+        }))
+        {
+            return true;
+        }
+        else
+        {
+            const availableSDKs = await this.getSDKs(existingPath);
+            if(availableSDKs.some((sdk) =>
+            {
+                // The SDK includes the Runtime, ASP.NET Core Runtime, and Windows Desktop Runtime. So, we don't need to check the mode.
+                return versionUtils.getMajorMinor(sdk.version, this.workerContext.eventStream, this.workerContext) === requestedMajorMinor;
+            }))
+            {
+                return true;
+            }
+
+            if(!this.allowInvalidPath(workerContext))
+            {
+                this.utilityContext.ui.showWarningMessage(`${badExistingPathWarningMessage}\nExtension: ${workerContext.acquisitionContext.requestingExtensionId ?? 'Unspecified'}`, () => {/* No Callback */}, );
+            }
+            return false;
+        }
+    }
+
+    private async getSDKs(existingPath : string) : Promise<IDotnetListInfo[]>
+    {
+        const findSDKsCommand = CommandExecutor.makeCommand(existingPath, ['--list-sdks']);
+
+        const sdkInfo = await (this.executor!).execute(findSDKsCommand).then((result) =>
+            {
+                const runtimes = result.stdout.split('\n').map((line) => line.trim()).filter((line) => line.length > 0);
+                const runtimeInfos : IDotnetListInfo[] = runtimes.map((sdk) =>
+                {
+                    if(sdk === '') // new line in output that got trimmed
+                    {
+                        return null;
+                    }
+                    const parts = sdk.split(' ', 2); // account for spaces in PATH, no space should appear before then and luckily path is last
+                    return {
+                        mode: 'sdk',
+                        version: parts[0],
+                        directory: sdk.split(' ').slice(1).join(' ').slice(1, -1) // need to remove the brackets from the path [path]
+                    } as IDotnetListInfo;
+                }).filter(x => x !== null) as IDotnetListInfo[];
```

Comment:
nit: This logic might be simpler as a RegEx, but I don't know the output being parsed. Just an idea.

Reply:
I can see the logic there, it would probably be shorter. But I am a bit opinionated on RegEx.  I would much rather debug code like this than a RegEx expression. I do appreciate this suggestion, though, and thanks for reviewing :)